tikv/raftstore/worker.go

```diff
@@ -295,25 +305,38 @@ func exceedEndKey(current, endKey []byte) bool {
 	return bytes.Compare(current, endKey) >= 0
 }
 
-/// splitCheck gets the split keys by scanning the range.
+/// SplitCheck gets the split keys by scanning the range.
 func (r *splitCheckHandler) splitCheck(startKey, endKey []byte, reader *dbreader.DBReader) [][]byte {
 	ite := reader.GetIter()
 	splitKeys := r.tryTableSplit(startKey, endKey, ite)
 	if len(splitKeys) > 0 {
 		return splitKeys
 	}
-	checker := newSizeSplitChecker(r.config.regionMaxSize, r.config.regionSplitSize, r.config.batchSplitLimit)
+	r.newCheckers()
+	abort := false
 	for ite.Seek(startKey); ite.Valid(); ite.Next() {
 		item := ite.Item()
 		key := item.Key()
 		if exceedEndKey(key, endKey) {
 			break
 		}
-		if checker.onKv(key, uint64(item.ValueSize())) {
+		for _, checker := range r.checkers {
```

Comment:
If we extract the for loop into a function, we don't need to define an `abort` variable before the outer loop.